website/src/content/docs/docs/howtos/Generate client libraries/02client.mdx

```diff
@@ -555,3 +555,77 @@ petActionClient.pet();
 ```
 
 </ClientTabs>
+
+### Adding Client Initialization Parameters
+
+By default, we only generate our clients with initialization parameters for `endpoint`, `credential`, and `apiVersion`, whenever any of these are applicable. There are cases where spec authors would like their clients to have additional input parameters. This is where the `@clientInitialization` decorator comes in.
```

Comment:
Why are implicit and not explicit? Should we make TSP authors be explicit and declare the client initiziation takes these three and any others?

Reply:
it's because these three parameters are required and can't be removed, and we already know what they are. We don't want to make users have to re-enter these 3 parameters. If we required users to input them in, we would raise a warning if the users don't input it, but to me it seems better to just include them automatically, since you can't remove them.

Reply:
Going to resolve so we can merge this docs pr, @RickWinter if you still feel this is an issue, can you open a separate one? thanks!

Reply:
Is the magic that the tsp compiler adds them? I wouldn't want to rely on emitters knowing to do this magic

Reply:
These are usually from different decorators in typespec source. For now, TCGC does not require dev explicitly write them in typespec clientInitialization (if it does, basically every single service need to add these bunch of parameters).

For example, `credential` be via `@useAuth`, e.g. https://github.com/weidongxu-microsoft/typespec/blob/java_migrate-specs/packages/http-specs/specs/authentication/api-key/main.tsp#L9
`endpoint` via `@server`, e.g. https://github.com/weidongxu-microsoft/typespec/blob/java_migrate-specs/packages/http-specs/specs/server/path/single/main.tsp#L9-L16
`api-version` is typically `Azure`, and from `@versioned`, e.g. https://github.com/weidongxu-microsoft/typespec/blob/java_migrate-specs/packages/http-specs/specs/versioning/added/main.tsp#L13